core/src/avm2/globals/flash/display/loader.rs

```diff
@@ -1,71 +1,91 @@
-pub fn class_init<'gc>(
-    _activation: &mut Activation<'_, 'gc, '_>,
-    _this: Option<Object<'gc>>,
-    _args: &[Value<'gc>],
-) -> Result<Value<'gc>, Error> {
-    Ok(Value::Undefined)
-}
-
-/// Implements `flash.display.Loader`'s instance constructor.
-pub fn instance_init<'gc>(
+pub fn init<'gc>(
     activation: &mut Activation<'_, 'gc, '_>,
     this: Option<Object<'gc>>,
     _args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error> {
-    if let Some(this) = this {
-        activation.super_init(this, &[])?;
+    if let Some(mut this) = this {
+        if this.as_display_object().is_none() {
+            let new_do = LoaderDisplay::new_with_avm2(activation.context.gc_context, this);
+            this.init_display_object(activation.context.gc_context, new_do.into());
+        }
+
+        // Some LoaderInfo properties (such as 'bytesLoaded' and 'bytesTotal') are always
+        // accessible, even before the 'init' event has fired. Using an empty movie gives
+        // us the correct value (0) for them.
+        let loader_info = LoaderInfoObject::not_yet_loaded(
+            activation,
+            Arc::new(SwfMovie::empty(activation.context.swf.version())),
+            Some(this),
+        )?;
+        this.set_property(
+            &QName::new(Namespace::private(""), "_contentLoaderInfo").into(),
+            loader_info.into(),
+            activation,
+        )?;
     }
 
     Ok(Value::Undefined)
 }
 
-// TODO: this is entirely stubbed, just to get addEventListener to not crash
-fn content_loader_info<'gc>(
-    _activation: &mut Activation<'_, 'gc, '_>,
+pub fn load<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
     this: Option<Object<'gc>>,
-    _args: &[Value<'gc>],
+    args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error> {
-    log::warn!("Loader::contentLoaderInfo is a stub");
-    Ok(this.unwrap().into())
-}
-
-pub fn create_class<'gc>(mc: MutationContext<'gc, '_>) -> GcCell<'gc, Class<'gc>> {
-    let class = Class::new(
-        QName::new(Namespace::package("flash.display"), "Loader"),
-        Some(
-            QName::new(
-                Namespace::package("flash.display"),
-                "DisplayObjectContainer",
-            )
-            .into(),
-        ),
-        Method::from_builtin(instance_init, "<Loader instance initializer>", mc),
-        Method::from_builtin(class_init, "<Loader class initializer>", mc),
-        mc,
-    );
+    if let Some(this) = this {
+        let url_request = args[0].as_object().unwrap();
 
-    let mut write = class.write(mc);
+        if let Some(context) = args.get(1) {
+            if !matches!(context, Value::Null) {
+                log::warn!(
+                    "Loader.load: 'context' argument is not yet implemented: {:?}",
+                    context
+                );
+            }
+        }
+        let url = url_request
+            .get_property(&QName::dynamic_name("url").into(), activation)?
```

Comment:
let's use `&Multiname::public("url")` in new code, this is the more direct construction

Reply:
I've adjusted this PR to use `&Multiname::public` where applicable.